client/my-sites/plans-v2/product-card/index.tsx

```diff
@@ -0,0 +1,180 @@
+const itemToCard = ( { type }: SelectorProduct ) => {
+	switch ( type ) {
+		case ITEM_TYPE_PLAN:
+			return JetpackPlanCard;
+		case ITEM_TYPE_BUNDLE:
+			return JetpackBundleCard;
+		case ITEM_TYPE_PRODUCT:
+		default:
+			return JetpackProductCard;
+	}
+};
+
+interface UpgradeNudgeProps {
+	item: SelectorProduct;
+	currencyCode: string;
+	onClick: PurchaseCallback;
+}
+
+const UpgradeNudgeWrapper = ( { item, currencyCode, onClick }: UpgradeNudgeProps ) => {
+	const upgradeToProductSlug =
+		getRealtimeFromDaily( item.costProductSlug || item.productSlug ) || '';
+	const selectorProductToUpgrade = slugToSelectorProduct( upgradeToProductSlug );
+
+	const isFetchingPrices = useSelector( ( state ) => isProductsListFetching( state ) );
+
+	const itemCost = useSelector( ( state ) => getProductCost( state, upgradeToProductSlug ) );
+	const monthlyItemCost = useSelector( ( state ) =>
+		getProductCost( state, selectorProductToUpgrade?.monthlyProductSlug || '' )
+	);
+
+	if ( ! isUpgradeable( item.productSlug ) || isFetchingPrices || ! selectorProductToUpgrade ) {
+		return null;
+	}
+
+	let originalPrice = 0;
+	let discountedPrice = undefined;
+	if ( ! isFetchingPrices && itemCost ) {
+		originalPrice = itemCost;
+		if ( monthlyItemCost && selectorProductToUpgrade.term !== TERM_MONTHLY ) {
+			originalPrice = monthlyItemCost * 12;
+			discountedPrice = itemCost;
+		}
+	}
+
+	return (
+		<JetpackProductCardUpgradeNudge
+			billingTimeFrame={ durationToText( selectorProductToUpgrade.term ) }
+			currencyCode={ currencyCode }
+			discountedPrice={ discountedPrice }
+			originalPrice={ originalPrice }
+			onUpgradeClick={ () => onClick( selectorProductToUpgrade ) }
+			selectorProduct={ selectorProductToUpgrade }
+		/>
+	);
+};
+
+interface ProductCardProps {
+	item: SelectorProduct;
+	onClick: PurchaseCallback;
+	siteId: number | null;
+	currencyCode: string;
+	className?: string;
+}
+
+const ProductCardWrapper = ( {
+	item,
+	onClick,
+	siteId,
+	currencyCode,
+	className,
+}: ProductCardProps ) => {
+	// Determine whether product is owned.
+	const sitePlan = useSelector( ( state ) => getSitePlan( state, siteId ) );
+	const siteProducts = useSelector( ( state ) => getSiteProducts( state, siteId ) );
+	const isOwned = useMemo( () => {
+		if ( sitePlan && sitePlan.product_slug === item.productSlug ) {
+			return true;
+		} else if ( siteProducts ) {
+			return siteProducts
+				.filter( ( product ) => ! product.expired )
+				.map( ( product ) => product.productSlug )
+				.includes( item.productSlug );
+		}
+		return false;
+	}, [ item.productSlug, sitePlan, siteProducts ] );
+
+	// Calculate the product price.
+	const isFetchingPrices = useSelector( ( state ) => isProductsListFetching( state ) );
+	const itemCost = useSelector( ( state ) =>
+		getProductCost( state, item.costProductSlug || item.productSlug )
+	);
+	const monthlyItemCost = useSelector( ( state ) =>
+		getProductCost( state, item.monthlyProductSlug || '' )
+	);
+
+	let originalPrice = 0;
```

Comment:
We use a similar piece of code in `UpgradeNudgeWrapper`. Maybe this could be extracted from the component.

Reply:
Good idea! Made a new React hook in 1dd141a0db0174b4c619ef02635cd1d39d9223bd.